src/commands/detectorDiagnostics.ts

```diff
@@ -0,0 +1,130 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from "vscode-azureextensionui";
+import { AppLensARMresponse, AppLensAPIResult } from './models/applensarmresponse';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as htmlhandlers from "handlebars";
+import { Errorable } from './utils/errorable';
+import ResourceManagementClient from 'azure-arm-resource/lib/resource/resourceManagementClient';
+
+const meta = require('../../package.json');
+
+export default async function detectorDiagnostics(
+    context: IActionContext,
+    target: any
+  ): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    if (cloudExplorer.available) {
+      const cloudTarget = cloudExplorer.api.resolveCommandTarget(target);
+
+      if (cloudTarget && cloudTarget.cloudName === "Azure" &&
+            cloudTarget.nodeType === "resource" && cloudTarget.cloudResource.nodeType === "cluster") {
+              const extensionPath = getExtensionPath();
+              if (extensionPath) {
+                loadClusterInsights(cloudTarget, extensionPath);
+              }
+        } else {
+          vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        }
+    }
+}
+
+function getExtensionPath(): string | undefined {
+  const publisherName = `${meta.publisher}.${meta.name}`;
+  const vscodeExtensionPath = vscode.extensions.getExtension(publisherName)?.extensionPath;
+
+  if (!vscodeExtensionPath) {
+    vscode.window.showInformationMessage('No Extension path found.');
+    return;
+  }
+  return vscodeExtensionPath;
+}
+
+function loadClusterInsights(
+  cloudTarget: k8s.CloudExplorerV1.CloudExplorerResourceNode,
+  extensionPath: string) {
+  const clsutername = cloudTarget.cloudResource.name;
+
+  vscode.window.withProgress({
+    location: vscode.ProgressLocation.Notification,
+    title: `Loading ${clsutername} diagnostics.`,
+    cancellable: false
+  }, () => {
+    return new Promise(async (resolve) => {
```

Comment:
Instead of `() => new Promise(...)` couldn't you just do `async () => { ... }`?  Not sure what the constraints are here.  (I have a wrapper called `longRunning` which allows me to write `longRunning("Doing stuff", () => someAsyncFunction())` - I think there's a version in `porter-vscode/src/utils/host.ts` otherwise hit me up for a pointer if it would be useful.)